File: trianginf.py
from numpy import zeros,array

def trianginf(T, b):
    # T est la matrice triangulaire inférieure
    # b est le vecteur second membre
    n = len(b)
    x = zeros(n)
    x[0] = b[0] / T[0][0] #b1 / T11
    for i in range(1, n):
        S = 0
        for k in range(0, i):
            S += T[i][k] * x[k]
        x[i] = (b[i] - S) / T[i][i]
    return x

# x=trianginf(array([[4,0,0],[3,-3,0],[-1,1,1]]), array([8,9,-2]))
# print(x)

File: julia/trianginf.jl
function triangInf(A::Matrix{T}, b::Vector{T}) where T <: Number
    n = size(A, 1)
    x = zeros(n)
    x[1] = b[1] / A[1, 1]
    for i = 2:n
        Σ = 0
        for j = 1:i
            Σ += A[i, j] * x[j]
        end
        x[i] = (b[i] - Σ) / A[i, i]
    end
    return x
end